### ISSUE TITLE
Racing conditions in JSON deserialization: cannot determine model class of name: <news_search>

### ISSUE THREAD

**Issue by zhemaituk** [dataforseo/JavaClient#4]
Reproducible in such scenario (dataforseo-client:1.0.18):
- Mock http response (so there is almost no wait of the http transport)
- Create `new SerpApi(apiClient)` and use it in multiple threads, call to `serpApi.googleNewsLiveAdvanced`. 4 to 8 threads usually suffice to reproduce.
- Run a few times until exception is thrown

It seems that the racing conditions happen somewhere during initialization. If exception does not happen soon after start - it keeps working fine.

Also performing such to force warmup before creating `new SerpApi(apiClient)` appears to help:
```
io.github.dataforseo.client.JSON.deserialize("""
                {
                    "tasks": [
                        {
                            "result": [
                                {
                                    "type": "news",
                                    "item_types": [
                                        "news_search"
                                    ],
                                    "items": [
                                        {
                                            "type": "news_search"
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }""", new TypeToken<SerpGoogleNewsLiveAdvancedResponseInfo>(){}.getType());
```

Exception:
```
java.lang.IllegalArgumentException: cannot determine model class of name: <news_search>
	at io.github.dataforseo.client.JSON.getClassByDiscriminator(JSON.java:1957) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.JSON.access$100(JSON.java:49) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.JSON$117.getClassForElement(JSON.java:886) ~[dataforseo-client-1.0.18.jar:na]
	at io.gsonfire.gson.TypeSelectorTypeAdapterFactory$TypeSelectorTypeAdapter.read(TypeSelectorTypeAdapterFactory.java:65) ~[gson-fire-1.9.0.jar:na]
	at io.gsonfire.gson.NullableTypeAdapter.read(NullableTypeAdapter.java:36) ~[gson-fire-1.9.0.jar:na]
	at io.gsonfire.util.JsonUtils$ConfigurableTypeAdapter.read(JsonUtils.java:99) ~[gson-fire-1.9.0.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:na]
	at io.gsonfire.util.JsonUtils.fromJsonTree(JsonUtils.java:72) ~[gson-fire-1.9.0.jar:na]
	at io.gsonfire.gson.HooksTypeAdapter.read(HooksTypeAdapter.java:53) ~[gson-fire-1.9.0.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:na]
	at io.github.dataforseo.client.model.GoogleNewsNewsSearchSerpElementItem$CustomTypeAdapterFactory$1.read(GoogleNewsNewsSearchSerpElementItem.java:389) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.model.GoogleNewsNewsSearchSerpElementItem$CustomTypeAdapterFactory$1.read(GoogleNewsNewsSearchSerpElementItem.java:354) ~[dataforseo-client-1.0.18.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:204) ~[gson-2.10.1.jar:na]
	at io.gsonfire.util.JsonUtils$ConfigurableTypeAdapter.read(JsonUtils.java:99) ~[gson-fire-1.9.0.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:na]
	at io.gsonfire.util.JsonUtils.fromJsonTree(JsonUtils.java:72) ~[gson-fire-1.9.0.jar:na]
	at io.gsonfire.gson.TypeSelectorTypeAdapterFactory$TypeSelectorTypeAdapter.read(TypeSelectorTypeAdapterFactory.java:82) ~[gson-fire-1.9.0.jar:na]
	at io.gsonfire.gson.NullableTypeAdapter.read(NullableTypeAdapter.java:36) ~[gson-fire-1.9.0.jar:na]
	at io.gsonfire.util.JsonUtils$ConfigurableTypeAdapter.read(JsonUtils.java:99) ~[gson-fire-1.9.0.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:na]
	at io.gsonfire.util.JsonUtils.fromJsonTree(JsonUtils.java:72) ~[gson-fire-1.9.0.jar:na]
	at io.gsonfire.gson.HooksTypeAdapter.read(HooksTypeAdapter.java:53) ~[gson-fire-1.9.0.jar:na]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.readIntoField(ReflectiveTypeAdapterFactory.java:212) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$FieldReflectionAdapter.readField(ReflectiveTypeAdapterFactory.java:433) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:393) ~[gson-2.10.1.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:na]
	at io.github.dataforseo.client.model.SerpGoogleNewsLiveAdvancedResultInfo$CustomTypeAdapterFactory$1.read(SerpGoogleNewsLiveAdvancedResultInfo.java:595) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.model.SerpGoogleNewsLiveAdvancedResultInfo$CustomTypeAdapterFactory$1.read(SerpGoogleNewsLiveAdvancedResultInfo.java:560) ~[dataforseo-client-1.0.18.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:204) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.readIntoField(ReflectiveTypeAdapterFactory.java:212) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$FieldReflectionAdapter.readField(ReflectiveTypeAdapterFactory.java:433) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:393) ~[gson-2.10.1.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:na]
	at io.github.dataforseo.client.model.SerpGoogleNewsLiveAdvancedTaskInfo$CustomTypeAdapterFactory$1.read(SerpGoogleNewsLiveAdvancedTaskInfo.java:503) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.model.SerpGoogleNewsLiveAdvancedTaskInfo$CustomTypeAdapterFactory$1.read(SerpGoogleNewsLiveAdvancedTaskInfo.java:468) ~[dataforseo-client-1.0.18.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:204) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.readIntoField(ReflectiveTypeAdapterFactory.java:212) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$FieldReflectionAdapter.readField(ReflectiveTypeAdapterFactory.java:433) ~[gson-2.10.1.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:393) ~[gson-2.10.1.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:299) ~[gson-2.10.1.jar:na]
	at io.github.dataforseo.client.model.SerpGoogleNewsLiveAdvancedResponseInfo$CustomTypeAdapterFactory$1.read(SerpGoogleNewsLiveAdvancedResponseInfo.java:465) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.model.SerpGoogleNewsLiveAdvancedResponseInfo$CustomTypeAdapterFactory$1.read(SerpGoogleNewsLiveAdvancedResponseInfo.java:430) ~[dataforseo-client-1.0.18.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:204) ~[gson-2.10.1.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1227) ~[gson-2.10.1.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1137) ~[gson-2.10.1.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1047) ~[gson-2.10.1.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1014) ~[gson-2.10.1.jar:na]
	at io.github.dataforseo.client.JSON.deserialize(JSON.java:4246) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.ApiClient.deserialize(ApiClient.java:939) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.ApiClient.handleResponse(ApiClient.java:1149) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.ApiClient.execute(ApiClient.java:1073) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.api.SerpApi.googleNewsLiveAdvancedWithHttpInfo(SerpApi.java:8234) ~[dataforseo-client-1.0.18.jar:na]
	at io.github.dataforseo.client.api.SerpApi.googleNewsLiveAdvanced(SerpApi.java:8215) ~[dataforseo-client-1.0.18.jar:na]
```

**Comment by lVics** [comment]
Thank you for noticing the issue. We have fixed it in the new version 1.0.19. Please check if this issue has disappeared for you

**Comment by zhemaituk** [comment]
Thank you for the quick turnaround, the issue is not reproducible anymore after upgrading to 1.0.19.
